src/matter/cluster/server/GroupsServer.ts

```diff
@@ -0,0 +1,124 @@
+/**
+ * @license
+ * Copyright 2022 The node-matter Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GroupsCluster, TlvAddGroupResponse } from "../GroupsCluster";
+import { ClusterServerHandlers } from "./ClusterServer";
+import { StatusCode } from "../../interaction/InteractionMessages";
+import { GroupId } from "../../common/GroupId.js";
+import { ObjectSchema } from "@project-chip/matter.js";
+
+/*
+TODO: Global Cluster fields needs to be added also here because, as discussed, based on the implementation.
+* Cluster Revision: 4
+* FeatureMap:
+  * Bit 0: Group Names - The ability to store a name for a group.
+* AttributeList:
+  * xxx
+* AcceptedCommandList:
+  * xxx
+* GeneratedCommandList: empty
+* EventList: empty
+* FabricIndex: empty
+ */
+
+/*
+TODO: If the Scenes server cluster is implemented on the same endpoint, the following extension field SHALL
+      be added to the Scene Table:
+      * OnOff
+ */
+
+/*
+TODO: Is a "groupcast" the groupId range 0xFF00 - 0xFFFF ??
+ */
+
+export const GroupsClusterHandler: () => ClusterServerHandlers<typeof GroupsCluster> = () => {
+    const clusterGroups = new Map<number, string>();
+
+    const addGroupLogic = (groupId: GroupId, groupName: string): TlvAddGroupResponse => {
```

Comment:
I think that way we can avoid code duplication when following that schema ... I just had no success to make the return type vorrect here (but hopefully easy one for you) :-)